### DISCUSSION TITLE
Enable Prefetching of Routes with Link component

### DISCUSSION THREAD

**Discussion by mrchrishahn** [expo/router#803] [Ideas]
Really loving the ease of routing that Expo Router adds. As someone coming from NextJS, I appreciate how familiar it feels. 
I wanted to suggest another feature that NextJS offers, which is the prefetching of routes using the Link component. 

Ideally, we'd want to create a perfectly seamless experience for our users navigating between screens, so it would be great if there were a way to run the data queries on the screen that they will navigate to next while they're still looking at the previous.

I'm currently considering implementing a wrapper around the Link component in which I explicitly prefetch the queries for the different data needs of the respective screen the Link is pointing towards. In my case, I'm using React Query to manage the server state. 

Would love to hear some feedback on whether such a feature may make it on the roadmap

**Comment by tslater** [comment]
Would also be nice to do this outside of the link, before the user interacts with the link (not sure if that was part of the proposal here).
